### ISSUE TITLE
Bayesian Optimization Meets Bayesian Optimal Stopping

### ISSUE THREAD

**Issue by sile** [sile/readings#28]
URL
-----

http://proceedings.mlr.press/v97/dai19a.html

Abstract
----------

> Bayesian optimization (BO) is a popular paradigm for optimizing the hyperparameters of machine learning (ML) models due to its sample efficiency. Many ML models require running an iterative training procedure (e.g., stochastic gradient descent). This motivates the question whether information available during the training process (e.g., validation accuracy after each epoch) can be exploited for improving the epoch efficiency of BO algorithms by early-stopping model training under hyperparameter settings that will end up under-performing and hence eliminating unnecessary training epochs. This paper proposes to unify BO (specifically, Gaussian process-upper confidence bound (GP-UCB)) with Bayesian optimal stopping (BO-BOS) to boost the epoch efficiency of BO. To achieve this, while GP-UCB is sample-efficient in the number of function evaluations, BOS complements it with epoch efficiency for each function evaluation by providing a principled optimal stopping mechanism for early stopping. BO-BOS preserves the (asymptotic) no-regret performance of GP-UCB using our specified choice of BOS parameters that is amenable to an elegant interpretation in terms of the exploration-exploitation trade-off. We empirically evaluate the performance of BO-BOS and demonstrate its generality in hyperparameter optimization of ML models and two other interesting applications.

Table of Contents
--------------------

- (1) Introduction
- (2) Background and Notations
  - (2-1) Bayesian Optimization (BO) and GP-UCB
  - (2-2) Bayesian Optimal Stopping (BOS)
- (3) BO-BOS Algorithm
  - (3-1) Algorithm Description
  - (3-2) BOS for Early Stopping in BO
- (4) Theoretical Analysis
- (5) Experiments and Discussion
  - (5-1) Hyperparameter Optimization of LR
  - (5-2) Hyperparameter Optimization of CNN
  - (5-3) Novel Applications of the BO-BOS Algorithm
    - (5-3-1) POLICY SEARCH FOR RL
    - (5-3-2) JOINT HYPERPARAMETER TUNING AND FEATURE SELECTION
- (6) Conclusion

BibTex
--------

```
@InProceedings{pmlr-v97-dai19a,
  title = 	 {{B}ayesian Optimization Meets {B}ayesian Optimal Stopping},
  author = 	 {Dai, Zhongxiang and Yu, Haibin and Low, Bryan Kian Hsiang and Jaillet, Patrick},
  booktitle = 	 {Proceedings of the 36th International Conference on Machine Learning},
  pages = 	 {1496--1506},
  year = 	 {2019},
  editor = 	 {Chaudhuri, Kamalika and Salakhutdinov, Ruslan},
  volume = 	 {97},
  series = 	 {Proceedings of Machine Learning Research},
  address = 	 {Long Beach, California, USA},
  month = 	 {09--15 Jun},
  publisher = 	 {PMLR},
  pdf = 	 {http://proceedings.mlr.press/v97/dai19a/dai19a.pdf},
  url = 	 {http://proceedings.mlr.press/v97/dai19a.html},
```

See Also
----------

- GitHub: https://github.com/daizhongxiang/Bayesian-Optimization-Meets-Bayesian-Optimal-Stopping

**Comment by sile** [comment]
> We focus on problem settings where the objective function `f` is bounded and monotonically increasing in `n`. (3)

**Comment by sile** [comment]
> ![image](https://user-images.githubusercontent.com/181413/63146830-4a12ef80-c037-11e9-9b27-3313bc7583fd.png)

Does this algorithm not use the results of early stopped trials? > "fixed at N"

**Comment by sile** [comment]
> when the recommended decision is `d[2]`, there is no early stopping and model training continues under `x[t]` for one more epoch.
